File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceConnectChannel.java
package com.lumiyaviewer.lumiya.voice.common.messages;

import android.os.Bundle;
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage;
import com.lumiyaviewer.lumiya.voice.common.model.VoiceChannelInfo;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public class VoiceConnectChannel implements VoicePluginMessage {

    @Nullable
    public final String channelCredentials;

    @Nonnull
    public final VoiceChannelInfo voiceChannelInfo;

    public VoiceConnectChannel(Bundle bundle) {
        this.voiceChannelInfo = new VoiceChannelInfo(bundle.getBundle("voiceChannelInfo"));
        this.channelCredentials = bundle.getString("channelCredentials");
    }

    public VoiceConnectChannel(@Nonnull VoiceChannelInfo voiceChannelInfo, @Nullable String channelCredentials) {
        this.voiceChannelInfo = voiceChannelInfo;
        this.channelCredentials = channelCredentials;
    }

    @Override
    public Bundle toBundle() {
        Bundle bundle = new Bundle();
        bundle.putBundle("voiceChannelInfo", this.voiceChannelInfo.toBundle());
        bundle.putString("channelCredentials", this.channelCredentials);
        return bundle;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceConnectChannel.kt
package com.lumiyaviewer.lumiya.voice.common.messages

import android.os.Bundle
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage
import com.lumiyaviewer.lumiya.voice.common.model.VoiceChannelInfo

class VoiceConnectChannel : VoicePluginMessage {
    @JvmField
    val channelCredentials: String?
    @JvmField
    val voiceChannelInfo: VoiceChannelInfo

    constructor(bundle: Bundle) {
        this.voiceChannelInfo = VoiceChannelInfo(bundle.getBundle("voiceChannelInfo"))
        this.channelCredentials = bundle.getString("channelCredentials")
    }

    constructor(voiceChannelInfo: VoiceChannelInfo, channelCredentials: String?) {
        this.voiceChannelInfo = voiceChannelInfo
        this.channelCredentials = channelCredentials
    }

    override fun toBundle(): Bundle = Bundle().apply {
        putBundle("voiceChannelInfo", voiceChannelInfo.toBundle())
        putString("channelCredentials", channelCredentials)
    }
}
